backend/src/main/java/gov/cdc/usds/simplereport/config/ApiUserAwareGraphQlContextBuilder.java

```diff
@@ -111,7 +119,23 @@ private DataLoaderRegistry buildDataLoaderRegistry() {
                           .map(p -> found.getOrDefault(p, PatientPreferences.DEFAULT))
                           .collect(Collectors.toList());
                     }));
-    dataLoaderRegistry.register(PatientPreferences.DATA_LOADER, patientPreferencesLoader);
+    dataLoaderRegistry.register(PatientPreferences.DATA_LOADER, patientPreferencesDataLoader);
+
+    DataLoader<UUID, TestEvent> testEventDataLoader =
+        new DataLoader<>(
+            patientIds ->
+                supplyAsync(
+                    () -> {
+                      Map<UUID, TestEvent> found =
+                          testEventRepository.findLastTestsByPatient(patientIds).stream()
+                              .collect(Collectors.toMap(TestEvent::getPatientInternalID, s -> s));
+                      return patientIds.stream()
+                          .map(te -> found.getOrDefault(te, null))
+                          .collect(Collectors.toList());
+                    }));
+    dataLoaderRegistry.register(PatientDataResolver.LAST_TEST_DATA_LOADER, testEventDataLoader);
+    dataLoaderRegistry.register(PatientPreferences.DATA_LOADER, patientPreferencesDataLoader);
```

Comment:
`register` is probably idempotent, but it's already called on line 122.
```suggestion
```

Reply:
Whoops, thanks!